src/Tools/NodeDocumentationMarkdownGenerator/Commands/FromDirectoryCommand.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using NodeDocumentationMarkdownGenerator.Verbs;
+
+namespace NodeDocumentationMarkdownGenerator.Commands
+{
+    internal static class FromDirectoryCommand
+    {
+        /// <summary>
+        /// Creates markdown files using the fromdirectory verb
+        /// </summary>
+        /// <param name="opts"></param>
+        internal static void HandleDocumentationFromDirectory(FromDirectoryOptions opts)
+        {
+            Program.VerboseMode = opts.Verbose;
+            var searchOption = opts.RecursiveScan ? SearchOption.AllDirectories : SearchOption.TopDirectoryOnly;
+            var fileInfos = new List<MdFileInfo>();
+            fileInfos.AddRange(ScanFolderForAssemblies(opts.InputFolderPath, opts.Filter, opts.ReferencePaths, searchOption));
+            if (opts.IncludeCustomNodes)
+            {
+                fileInfos.AddRange(ScanFolderForCustomNodes(opts.InputFolderPath, searchOption));
+            }
+
+            MarkdownHandler.CreateMdFilesFromFileNames(fileInfos, opts.OutputFolderPath, opts.Overwrite, opts.CompressImages, opts.CompressGifs, opts.DictionaryDirectory, opts.LayoutSpecPath);
+        }
+
+        private static List<MdFileInfo> ScanFolderForCustomNodes(string inputFolderPath, SearchOption searchOption)
+        {
+            var allDyfs = Directory.GetFiles(inputFolderPath, "*.dyf", searchOption).Select(x => new FileInfo(x)).ToList();
+            var fileInfos = new List<MdFileInfo>();
+            foreach (var cn in allDyfs)
+            {
+                var fileInfo = MdFileInfo.FromCustomNode(cn.FullName);
+                if (fileInfo is null) continue;
+
+                fileInfos.Add(fileInfo);
+            }
+            return fileInfos;
+        }
+
+        private static List<MdFileInfo> ScanFolderForAssemblies(string inputFolderPath, IEnumerable<string> filter, IEnumerable<string> referencePaths, SearchOption searchOption)
+        {
+            var allAssembliesFromInputFolder = Directory.GetFiles(inputFolderPath, "*.*", searchOption)
+                .Where(x => x.EndsWith(".dll") || x.EndsWith(".ds"))
+                .Select(x => new FileInfo(x))
+                .GroupBy(x => x.Name)
+                .Select(x => x.FirstOrDefault())
+                .ToList();
```

Comment:
Any reason to force all the IEnumerables into Lists this early ?

Reply:
let me look at this, it does seem a bit odd since we can just do it once at the end of the method.